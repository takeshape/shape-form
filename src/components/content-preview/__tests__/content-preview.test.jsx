import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ReactShallowRenderer();
import ContentPreview from '../content-preview';

describe('ContentPreview', () => {
  it('renders', () => {
    const props = {
      value: 'Andrew',
      schema: {type: 'string', title: 'Name'},
      formName: 'test-form',
      path: 'name',
      config: {},

      onChange: jest.fn(),
      onChangeSilent: jest.fn(),
      onBlur: jest.fn(),
      onFocus: jest.fn()
    };

    const tree = renderer.render(<ContentPreview {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders custom widget', () => {
    const props = {
      value: 'Andrew',
      schema: {type: 'string', title: 'Name'},
      formName: 'test-form',
      path: 'name',
      config: {preview: 'customPreview'},
      widgets: {
        customPreview({value}) { // eslint-disable-line
          return <div>{value}</div>;
        }
      },

      onChange: jest.fn(),
      onChangeSilent: jest.fn(),
      onBlur: jest.fn(),
      onFocus: jest.fn()
    };

    const tree = renderer.render(<ContentPreview {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('falls back on DefaultContentPreview if the configured widget is undefined', () => {
    const props = {
      value: 'Andrew',
      schema: {type: 'string', title: 'Name'},
      formName: 'test-form',
      path: 'name',
      config: {},
      widgets: {},

      onChange: jest.fn(),
      onChangeSilent: jest.fn(),
      onBlur: jest.fn(),
      onFocus: jest.fn()
    };

    const tree = renderer.render(<ContentPreview {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
