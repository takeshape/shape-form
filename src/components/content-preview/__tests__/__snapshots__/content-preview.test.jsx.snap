// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPreview falls back on DefaultContentPreview if the configured widget is undefined 1`] = `
<DefaultContentPreview
  config={Object {}}
  formName="test-form"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  onChangeSilent={[MockFunction]}
  onFocus={[MockFunction]}
  path="name"
  schema={
    Object {
      "title": "Name",
      "type": "string",
    }
  }
  value="Andrew"
  widgets={Object {}}
/>
`;

exports[`ContentPreview renders 1`] = `
<DefaultContentPreview
  config={Object {}}
  formName="test-form"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  onChangeSilent={[MockFunction]}
  onFocus={[MockFunction]}
  path="name"
  schema={
    Object {
      "title": "Name",
      "type": "string",
    }
  }
  value="Andrew"
/>
`;

exports[`ContentPreview renders custom widget 1`] = `
<customPreview
  config={
    Object {
      "preview": "customPreview",
    }
  }
  formName="test-form"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  onChangeSilent={[MockFunction]}
  onFocus={[MockFunction]}
  path="name"
  schema={
    Object {
      "title": "Name",
      "type": "string",
    }
  }
  value="Andrew"
  widgets={
    Object {
      "customPreview": [Function],
    }
  }
/>
`;
